### PR TITLE
[CMake][SofaKernel] FIX: exporting options in SofaFrameworkConfig.cmake

### PR DESCRIPTION
**Pull request by marques-bruno**
Hi!

When building plugins as standalone projects, the SofaFramework package doesn't expose the cmake config options used to build SOFA.
In the SoftRobots plugin (among others), tests are performed to know if SOFA has been built using the required options (such as SOFA_BUILD_METIS, MODULE_SOFASPARSESOLVER, PLUGIN_SOFAPYTHON, SOFA_BUILD_TESTS etc.) and out-of-tree configurations require to re-specify those options, even though they are only defined in the SOFA build.

I've modified the SofaFrameworkConfig.cmake.in to expose those options to whoever does a find_package(SofaFramework), although that might not be the smartest way to do it (the list is non exhausive and it would be best if options were added in a smarter way.
I think that modules such as SOFASPARSESOLVER should export the MODULE_SOFASPARSESOLVER themselves for instance, and so should the plugins.
But then there's also a long list of options that would be worth exposing, such as:
- SOFA_BUILD_TESTS
- SOFA_BUILD_METIS
- SOFA_BUILD_TUTORIALS
- SOFA_WITH_DEPRECATED_COMPONENTS
- SOFA_WITH_THREADING
- SOFA_WITH_DEVTOOLS
- SOFA_USE_MASK
- ....

And I don't know where those should be exposed.. So I'm looking fwd to your opinion on the question!

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

**Comment by damienmarchal** [comment]
Hello Bruno,

Thanks for the work done, I fully agree that this is needed. 

I don't understand the need for MODULE_SOFASPARSESOLVER. This is a module/plugin so is the classic find_package(SofaSparseSolver) then (SofaSparseSolver_FOUND) not enough ?
It should be the same with SofaPython. Its availability should be tested with find_package(SofaPython). 

Now about the other defines that are not related to module or plugin, it is of course important to export them properly in the SofaFramework package.

File: SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in [hunk 1]
```diff
@@ -17,6 +17,16 @@ set(SOFA_HAVE_GLEW "@SOFA_HAVE_GLEW@")
 set(SOFA_HAVE_BOOST "@SOFA_HAVE_BOOST@")
 set(SOFA_NO_OPENGL "@SOFA_NO_OPENGL@")
 set(SOFA_HAVE_GTEST "@SOFA_HAVE_GTEST@")
+set(SOFA_BUILD_TESTS "@SOFA_BUILD_TESTS@")
+
+set(SOFA_BUILD_METIS "@SOFA_BUILD_METIS@")
+set(PLUGIN_SOFAPYTHON "@PLUGIN_SOFAPYTHON@")
```
Comment (damienmarchal) [inline SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in]: Why not just a find_package(SofaPython)

File: SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in [hunk 2]
```diff
@@ -17,6 +17,16 @@ set(SOFA_HAVE_GLEW "@SOFA_HAVE_GLEW@")
...
+set(SOFA_USE_MASK "@SOFA_USE_MASK@")
+set(SOFA_WITH_DEVTOOLS "@SOFA_WITH_DEVTOOLS@")
+set(SOFA_WITH_THREADING "@SOFA_WITH_THREADING@")
+set(SOFA_WITH_DEPRECATED_COMPONENTS "@SOFA_WITH_DEPRECATED_COMPONENTS@")
+set(SOFA_BUILD_TUTORIALS "@SOFA_BUILD_TUTORIALS@")
+set(MODULE_SOFASPARSESOLVER "@MODULE_SOFASPARSESOLVER@")
```
Comment (damienmarchal) [inline SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in]: I'm not sure how precise it is needed but what about
```cmake
find_package(SofaSparseSolver)
if(SofaSparseSolver_FOUND)
      if(SofaSparseSolver_OTHEROPTIONS)
                 ....
     endif()
endif()
```

Comment (marques-bruno) [reply SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in]: Very good point... and obvious answer.
I was looking at how SoftRobots and STLIB looks for its dependencies, and didn't think of replacing the search for the CMake option "MODULE_SOFAXXX" and "PLUGIN_XXX" with the "XXX_FOUND" generated in the find_package...

For all the other cases (SOFA_WITH_*, SOFA_BUILD_*, SOFA_USE_*, ...), are you OK with exporting them in the SofaFrameworkConfig.cmake?

**Comment by damienmarchal** [comment]
Hello bruno, 

Is this PR still in WIP state ? 

Regards

File: SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in (hunk 2 above)
Comment (marques-bruno) [reply SofaKernel/SofaFramework/SofaFrameworkConfig.cmake.in]: While you're right that it's how the module import should be done, a find_package(SofaSparseSolver) in an out-of-tree plugin build results in the following error:

```
CMake Error at CMakeLists.txt:11 (find_package):
  Found package configuration file:

    /home/bruno/dev/SOFA/sofa/build/install/lib/cmake/SofaSparseSolver/SofaSparseSolverConfig.cmake

  but it set SofaSparseSolver_FOUND to FALSE so package "SofaSparseSolver" is
  considered to be NOT FOUND.  Reason given by package:

  The following imported targets are referenced, but are missing:
  SofaBaseLinearSolver SofaGeneralLinearSolver SofaImplicitOdeSolver
  SofaSimpleFem metis csparse
```

It looks like the module's dependency targets are not properly referenced. I'll investigate, either in this PR or in another one..(?)
Other than that I removed the PLUGIN_SOFAPYTHON variable export & to me this PR is ready for review

**Comment by marques-bruno** [comment]
As far as I tested, it works :+1:
-> ready imho

**Pull request merged**